lib/jekyll/readers/theme_assets_reader.rb

```diff
@@ -0,0 +1,47 @@
+module Jekyll
+  class ThemeAssetsReader
+    attr_reader :site
+    def initialize(site)
+      @site = site
+    end
+
+    def read
+      return unless site.theme && site.theme.assets_path
+
+      Find.find(site.theme.assets_path) do |path|
+        next if File.directory?(path)
+        if File.symlink?(path)
+          Jekyll.logger.warn "Theme reader:", "Ignored symlinked asset: #{path}"
+        else
+          read_theme_asset(path)
+        end
+      end
+    end
+
+    private
+    def read_theme_asset(path)
+      base = site.theme.root
+      dir = File.dirname(path.sub("#{site.theme.root}/", ""))
```

Comment:
We do this same thing at least 2 different ways elsewhere, in the `Jekyll::Layout` we use the equivalent of `path.sub(site.theme.root, "")` and in the `Jekyll::Document` we use `Pathutil` and there are probably more ways we're doing this. :confounded: